File: backend/Models/Book.cs
using System.Collections.Generic;
using System;

namespace backend.Models;

public class Book
{
    public int Id { get; set; }

    public string Title { get; set; } = string.Empty;

    public string Author { get; set; } = string.Empty;

    public int Year { get; set; }

    public string? Description { get; set; }

    public bool IsAvailable { get; set; } = true;

    public DateTime? ReturnDueDate { get; set; }

    public virtual ICollection<Borrowing>? Borrowings { get; set; }
}

File: frontend/src/types/book.ts
export interface GetBooksParams {
  title?: string;
  author?: string;
  isAvailable?: boolean;
  sortBy?: string;
  sortDir?: "asc" | "desc";
  page?: number;
  pageSize?: number;
}

export interface Book {
  id: number;
  title: string;
  author: string;
  year: number;
  description: string | null;
  isAvailable: boolean;
  returnDueDate: string | null;
}

export interface PagedResult<T> {
  items: T[];
  totalCount: number;
}

export interface BookFiltersValues {
  title?: string;
  author?: string;
  isAvailable?: string;
  sortBy?: string;
  sortDir?: "asc" | "desc";
}

export interface BookBorrowing {
  username: string;
  borrowedAt: string;
  returnDueDate: string;
  returnedAt: string | null;
}
